src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,28 +24,153 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
-    }
 
-    private void refreshTestData(String filename) {
-        testData = loadDataBundle(DATA_FOLDER_PATH + filename);
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            increaseNumOfStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            increaseNumOfStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            increaseNumOfQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            increaseNumOfQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
         removeAndRestoreDataBundle(testData);
     }
 
+    private void increaseNumOfStudents(int numStudentsToAdd) {
+        int countLimit = 0;
+        for (String key : testDataMax.students.keySet()) {
+            if (countLimit >= numStudentsToAdd) {
+                return;
+            }
+            if (!testData.students.containsKey(key)) {
+                testData.students.put(key, testDataMax.students.get(key));
+                countLimit++;
+            }
+        }
+    }
+
+    private void decreaseNumOfStudents(int remainingNumOfStudents) {
```

Comment:
Consider renaming this method: its similarity to `increaseNumOfStudents` suggests it decreases the number of students by the integer passed in.